### PR TITLE
Feature rule dates

### PR DESCRIPTION
**Pull request by deepaksingh1308**
- Integrated date-effective and date-expires in DRL rules
- Refactored Drools Rule Model to have those fields.
- Removed isActive check before loading of rules as rules now work on drl dates.
- Updated API Request object to take dates as inputs from UI

### CODE REVIEW

File: src/main/java/com/wf/stp/rbms/service/UpoCreateRulesService.java [hunk 1]
```diff
@@ -49,8 +50,14 @@ public boolean saveRules(CreateRulesRequest createRulesRequest) {
         droolsRuleModel.setUpoRuleThenList(createRulesRequest.getUpoRuleThenStatementList());
         droolsRuleModel.setRuleWhenPart(ruleWhenPart);
         droolsRuleModel.setRuleThenPart(ruleThenPart);
-        droolsRuleModel.setCreateTimeStamp(LocalDateTime.now().toString());
-        droolsRuleModel.setEndTimeStamp(LocalDateTime.parse("2099-12-31T23:59:59.999999999").toString());
+        droolsRuleModel.setRuleStartDate(createRulesRequest.getRuleStartDate());
+
+        if (Objects.nonNull(createRulesRequest.getRuleEndDate()) && !createRulesRequest.getRuleEndDate().isEmpty()) {
```
Comment (rkpandita) [inline src/main/java/com/wf/stp/rbms/service/UpoCreateRulesService.java:55]: StringUtils.isEmpty() checks for both null and length cases, please check if that can be used instead.

File: src/main/java/com/wf/stp/rbms/service/UpoDroolsService.java [hunk 2]
```diff
@@ -36,12 +36,9 @@ public class UpoDroolsService {
     private void createDRLFilePostConstruct() {
         final Map<String, String> serviceToDRLFileMap = new HashMap<>();
 
-        RbmsUTIL.getUpoFieldMap().keySet()
-                .parallelStream()
-                .forEach(service -> {
+        RbmsUTIL.getUpoFieldMap().keySet().parallelStream().forEach(service -> {
             final StringBuffer str = new StringBuffer();
```
[the remark this answers is not in the document]
Comment (deepaksingh1308) [reply src/main/java/com/wf/stp/rbms/service/UpoDroolsService.java:40]: Unfortunately, can't be done. This string buffer is to create service specific DRL file. If we move it outside there will be an overlap among DRLs.

**Review by rkpandita — APPROVED** [review]
Looks good to me.

File: src/main/java/com/wf/stp/rbms/service/UpoDroolsService.java [hunk 3]
```diff
@@ -36,12 +36,9 @@ public class UpoDroolsService {
...
             final Query query = new Query();
-            query.addCriteria(Criteria.where("isActive").is(true));
             query.addCriteria(Criteria.where("serviceName").is(service));
```
Comment (YogeshPitale) [inline src/main/java/com/wf/stp/rbms/service/UpoDroolsService.java:42]: I guess isActive check is redundant once we add start and end dates?

Comment (deepaksingh1308) [reply src/main/java/com/wf/stp/rbms/service/UpoDroolsService.java:42]: Yes. That's the reason I removed it.
This line is removed, not added. :)

**Pull request merged**